lib/super_digit.rb

```diff
@@ -1,15 +1,29 @@
 # Superdigit
 
-# Time Complexity - ?
-# Space Complexity - ?
+# Time Complexity - O(n) - n is the length of the number, but there is also a consideration for the value of those numbers (9999 will need more calculating than 1000)
+# Space Complexity - O(1) - space stays constant as n changes
 def super_digit(n)
```

Comment:
The way you're doing it looks a little strange, but it does work, nice and it avoids any recursion.